src/pages/Account/LoginWithCustomToken.tsx

```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import { useFirebase } from "react-redux-firebase";
+import { useHistory, useParams } from "react-router-dom";
+import { useAsync } from "react-use";
+
+import { isDefined } from "utils/types";
+import { enterVenue } from "utils/url";
+
+import { useUser } from "hooks/useUser";
+
+import { LoadingPage } from "components/molecules/LoadingPage";
+
+import "./Account.scss";
+
+export interface LoginCustomCodeProps {}
+
+export const LoginWithCustomToken: React.FC<LoginCustomCodeProps> = () => {
+  const { venueId, customToken } = useParams<{
+    venueId?: string;
+    customToken?: string;
+  }>();
+
+  const { user } = useUser();
+
+  const isLoggedInUser = isDefined(user);
+
+  const firebase = useFirebase();
+
+  const { replace: replaceUrlUsingRouter } = useHistory();
+
+  const { loading: isCustomTokenLoginLoading, error } = useAsync(async () => {
+    if (!venueId || !customToken)
+      throw new Error("venueId and customToken are required");
+
+    if (isLoggedInUser) throw new Error("there is already a logged in user");
+
+    // @debt: move this into api/auth or similar?
+    await firebase
+      .auth()
+      .signInWithCustomToken(customToken)
+      // .catch((error) => {
+      //   // TODO: Handle this with Bugsnag or similar?
+      //   console.error(error);
+      //
+      //   throw error;
+      // });
+      .then(() => {
+        enterVenue(venueId, { customOpenRelativeUrl: replaceUrlUsingRouter });
+      });
```

Comment:
[TODO] move this into api/auth or similar? Add Bugsnag or cleanup commented part.